File: Burp/checkquestions.py
import os
import json
from collections import defaultdict

OUTPUT_DIR = "output"

def count_questions_in_file(filepath):
    try:
        with open(filepath, "r", encoding="utf-8") as f:
            data = json.load(f)
            return len(data)
    except Exception as e:
        print(f"❌ Error reading {filepath}: {e}")
        return None

def main():
    question_bank_totals = defaultdict(int)
    file_counts = []

    for filename in sorted(os.listdir(OUTPUT_DIR)):
        if filename.endswith(".json"):
            filepath = os.path.join(OUTPUT_DIR, filename)
            count = count_questions_in_file(filepath)
            if count is not None:
                file_counts.append((filename, count))
                bank_name = filename.split("_questions")[0]
                question_bank_totals[bank_name] += count

    print("\n🧮 Total questions by bank (by file):\n")
    for bank_name in sorted(question_bank_totals):
        print(f"{bank_name}: {question_bank_totals[bank_name]} questions")

    # Combine weightings into base bank categories
    combined_totals = defaultdict(int)
    for bank_name, count in question_bank_totals.items():
        base_bank = (
            bank_name.replace("_low", "")
                     .replace("_medium", "")
                     .replace("_high", "")
                     .replace("_questions", "")
        )
        combined_totals[base_bank] += count

    print("\n📦 Combined total questions per bank:\n")
    for base_bank in sorted(combined_totals):
        print(f"{base_bank}: {combined_totals[base_bank]} questions")

if __name__ == "__main__":
    main()

File: Burp/checkquestions.js
#!/usr/bin/env node
const fs = require('fs/promises');
const path = require('path');

const OUTPUT_DIR = 'question_banks';

async function countQuestionsInFile(filepath) {
  try {
    const code = await fs.readFile(filepath, 'utf-8');
    const mod = await import('data:text/javascript,' + encodeURIComponent(code));
    const data = mod.default;
    return Array.isArray(data) ? data.length : 0;
  } catch (err) {
    console.error(`❌ Error reading ${filepath}: ${err}`);
    return null;
  }
}

async function main() {
  const questionBankTotals = {};
  const fileCounts = [];

  const files = (await fs.readdir(OUTPUT_DIR)).sort();
  for (const filename of files) {
    if (filename.endsWith('.js')) {
      const filepath = path.join(OUTPUT_DIR, filename);
      const count = await countQuestionsInFile(filepath);
      if (count !== null) {
        fileCounts.push([filename, count]);
        const bankName = filename.split('_questions')[0];
        questionBankTotals[bankName] = (questionBankTotals[bankName] || 0) + count;
      }
    }
  }

  console.log('\n🧮 Total questions by bank (by file):\n');
  const sortedBanks = Object.keys(questionBankTotals).sort();
  for (const bankName of sortedBanks) {
    console.log(`${bankName}: ${questionBankTotals[bankName]} questions`);
  }

  const combinedTotals = {};
  for (const [bankName, count] of Object.entries(questionBankTotals)) {
    const baseBank = bankName
      .replace('_low', '')
      .replace('_medium', '')
      .replace('_high', '')
      .replace('_questions', '');
    combinedTotals[baseBank] = (combinedTotals[baseBank] || 0) + count;
  }

  console.log('\n📦 Combined total questions per bank:\n');
  const sortedCombined = Object.keys(combinedTotals).sort();
  for (const baseBank of sortedCombined) {
    console.log(`${baseBank}: ${combinedTotals[baseBank]} questions`);
  }
}

if (require.main === module) {
  main();
}
